chat/be/server.js

```diff
@@ -175,13 +175,29 @@ async function run() {
       { $inc: { unreadCount: 1 } }
     );
 
-    io.emit("message", newMessage);
+    io.emit('message', newMessage);
 
     res.sendStatus(200);
   });
 
-  app.get("*", (req, res) => {
-    res.sendFile(__dirname + "/public/index.html");
+  app.delete('/api/rooms/:id/messages', async (req, res, next) => {
+    const roomId = req.params.id;
+    const messageId = req.body.id;
+
+    await db
+      .collection('messages')
+      .deleteOne({ _id: mongodb.ObjectId(messageId) });
+
+    io.emit('delete-message', {
```

Comment:
I'm sure you already know, but this has the same problem as the the new message in that we emit this even to people that don't have access to the room.  This is less worrisome for the delete message, but the new message one allows messages to be read by people that should not have access which is no good.  I've added a note to the Backlog for this.

Reply:
Oh from the docs I see something like this? `socket.to("room1").emit(/* ... */);`